types/reactstrap/reactstrap-tests.tsx

```diff
@@ -281,7 +281,10 @@ const Example13 = (
 interface CustomButtonProps extends ButtonProps {
   customProp: string;
 }
-const CustomButton: React.SFC<CustomButtonProps> = props => <Button {...props} />;
+// NOTE: not adding the <{}> causes the generic parameter to be a spread type of CustomButtonProps,
+// for some reason this causes children to be inferred as being 'ReactNode & {}' which makes the spread
+// invalid. TS3.2 bug?
+const CustomButton: React.SFC<CustomButtonProps> = props => <Button<{}> {...props}/>;
```

Comment:
Even just `props => <Button>{props.children}</Button>` would cause crazy build errors but _only_ in dtslint. The error wouldn't be flagged by tsservice inside vscode even with the version correctly set to use node_modules'.

```
ERROR: 287:62  expect  TypeScript@next compile error:
Type '{ children: ReactNode; }' is not assignable to type 'Readonly<ButtonProps<{ children: {}; }>>'.
  Types of property 'children' are incompatible.
    Type 'ReactNode' is not assignable to type '{} | ReactElement<any> | ReactNodeArray | ReactPortal | (string & {}) | (number & {}) | (false & {}) | (true & {})'.
      Type 'undefined' is not assignable to type '{} | ReactElement<any> | ReactNodeArray | ReactPortal | (string & {}) | (number & {}) | (false & {}) | (true & {})'.
```

Reply:
Hm. It *looks* like the SFC definition of `children` (it intersects it with the passed props to form the argument type in the react .d.ts) is incompatible with the `children` of `CustomButtonProps`. I'd be interested in looking at this a bit more. It might be that inference is failing for some reason. Would you mind opening an issue on the TS repo to track it, so I remember to look at it soon even if you merge this to get DT's react PRs flowing again?

Reply:
It's really weird, because both SFC's definition and ButtonProps' definitions are `children?: ReactNode`.

Reply:
Hm. Yeah, definitely sounds odd, at least. The "`{}` instead of the written value" description sounds a lot like an inference failure of some sort. Since it could be indicative of a flaw in our inference, it bears looking into if reproducible.

Reply:
Yeah, it still happened with today's nightly but again **only** in dtslint; I couldn't get it to happen at all with "use workspace version `3.2.0-dev.20181107`" inside vscode.

I'll make an issue in a bit.